notebooks/1.2-pyc-eda-mix.py

```diff
@@ -0,0 +1,500 @@
+#!/usr/bin/env python
+# coding: utf-8
+
+# # EDA MIX data
+# 
+# To explore all the 81 columns of this data frame the NEO team decide to split the EDA into 3 notebooks where:
+# 1- EDA of Structural Data
+# 2- EDA of Rooms data
+# 3- EDA of Mix Data (data that's not about Structural and Rooms data)
+# 
+# This notebook had the role to analyze the MIX data, **focus on the distribution of the variable, and try to get some useful insights**.
+
+from plotly.subplots import make_subplots
+import plotly.graph_objects as go
+import plotly.express as px
+import seaborn as sns
+import pandas as pd
+import numpy as np
+
+#Cdf function
+def ecdf(x):
+    x = np.sort(x)
+    def result(v):
+        return np.searchsorted(x, v, side='right') / x.size
+    return result
+
+def multiple_ecdf(column, target='SalePrice'):
+    fig = go.Figure()
+    keys = train[column].unique()
+    for key in keys:
+        bool_series = train[column]==key
+        fig.add_trace(
+            go.Scattergl(
+                x=np.unique(train[bool_series][target]), 
+                y=ecdf(train[bool_series][target])(np.unique(train[bool_series][target])), 
+                line_shape='hv',
+                name=str(key) + ', total: ' + str(bool_series.sum()))
+        )
+    return fig.show()
+
+train = pd.read_csv('../data/raw/train.csv')
+
+
+# ## Mix Columns overview
+# 
+# The columns of the training dataset that represents what we're calling mix data are represented in the strings in the variable **mix_columns**
+
+mix_columns = ['MSSubClass', 'MSZoning', 'LotFrontage', 'LotArea',
+               'Street', 'Alley', 'LotShape', 'LandContour', 'Utilities',
+               'LotConfig', 'LandSlope', 'Neighborhood', 'Condition1',
+               'Condition2', 'YearBuilt', 'YearRemodAdd', 'BldgType', 'HouseStyle',
+               'Functional', 'MiscFeature', 'MiscVal', 'MoSold', 'YrSold',
+               'SaleType', 'SaleCondition', 'SalePrice']
+
+train[mix_columns].head()
+
+
+train[mix_columns].isna().mean()
+
+
+train[mix_columns].describe()
+
+
+# # Sales variables
+# 
+# - MoSold
+# - YrSold
+# - SaleType
+# - SaleCondition
+
+# ## Sale Condition
+# 
+# 
+# Condition of sale
+# 
+#        Normal	Normal Sale
+#        Abnorml	Abnormal Sale -  trade, foreclosure, short sale
+#        AdjLand	Adjoining Land Purchase
+#        Alloca	Allocation - two linked properties with separate deeds, typically condo with a garage unit	
+#        Family	Sale between family members
+#        Partial	Home was not completed when last assessed (associated with New Homes)
+# 
+
+multiple_ecdf('SaleCondition')
+
+
+# **Insights**
+# 
+# - The partial category in this column appears to increase significantly the sale Price
```

Comment:
Is the number of data enough? If yes, so ok
Should we join and give the same relevance for them ? like change the values for just one ?

Reply:
I don't know. It's just a suggestion, because this values increase a lot the sale price.